### ISSUE TITLE
Annotations being added that cause "Unknown alert state" and blanking of graph

### ISSUE THREAD

**Issue by mbentley** [grafana/grafana#51555]
**What happened**:

This may be related to unified alerting but just recently, I've found that there are annotations that are being added where the state that is being set is not known.  In the UI when trying to mouse over them, the user receives an error and the graph blanks out until refreshed:

![unknown_alert_state](https://user-images.githubusercontent.com/414445/176443555-a80fe76e-92de-4f2d-a1d2-87d6e72cfa83.gif)

Looking at the annotations in the database, I can see there are some states that aren't valid where `new_state` has `Alerting (Error)` and `Pending (Error)`:
![screenshot 2022-06-29 at 9 06 02 AM](https://user-images.githubusercontent.com/414445/176443633-d29949bc-f5e1-45a4-abf1-513d1a959537.png)

Both of these times line up with two annotations in my dashboard that both trigger this behavior.

I can see a few other instances of what I can see cause the same behavior in other panels:

```
SELECT
    *
FROM
    `annotation`
WHERE
    `new_state` LIKE "%(Error)%"
ORDER BY `annotation`.`epoch` DESC;
```

```
id	org_id	alert_id	user_id	dashboard_id	panel_id	category_id	type	title	text	metric	prev_state	new_state	data	epoch   	region_id	tags	created	updated	epoch_end	
4238	1	52	0	13	60	NULL			UPS Status (Charge) alert {alertname=UPS Status (C...	NULL	Pending (Error)	Alerting (Error)		1656465785169	0	[]	1656465904054	1656465904054	1656465785169	
4244	1	47	0	13	54	NULL			Rack Overview - Ping Stats (Xfinity Only) alert {a...	NULL	Pending (Error)	Alerting (Error)		1656465785169	0	[]	1656465904243	1656465904243	1656465785169	
4248	1	41	0	13	2	NULL			Rack Overview - Ping Stats alert {alertname=Rack O...	NULL	Pending (Error)	Alerting (Error)		1656465785169	0	[]	1656465904734	1656465904734	1656465785169	
4255	1	48	0	8	49	NULL			athena.casa.mbentley.net - Out of Memory Events al...	NULL	Pending (Error)	Alerting (Error)		1656465785169	0	[]	1656465907890	1656465907890	1656465785169	
4194	1	41	0	13	2	NULL			Rack Overview - Ping Stats alert {alertname=Rack O...	NULL	Normal	Pending (Error)		1656445475169	0	[]	1656445475331	1656445475331	1656445475169	
4195	1	47	0	13	54	NULL			Rack Overview - Ping Stats (Xfinity Only) alert {a...	NULL	Normal	Pending (Error)		1656445475169	0	[]	1656445480264	1656445480264	1656445475169	
4196	1	48	0	8	49	NULL			athena.casa.mbentley.net - Out of Memory Events al...	NULL	Normal	Pending (Error)		1656445475169	0	[]	1656445482756	1656445482756	1656445475169	
4197	1	52	0	13	60	NULL			UPS Status (Charge) alert {alertname=UPS Status (C...	NULL	Alerting	Alerting (Error)		1656445475169	0	[]	1656445494758	1656445494758	1656445475169	
```

Looking at the code, it seems like this isn't valid:
https://github.com/grafana/grafana/blob/v9.0.2/public/app/features/alerting/state/alertDef.ts#L87-L157

I believe this is why it is showing up as an `Unknown alert state`

**What you expected to happen**:

Better handling of unknown states in annotations and/or alerts not setting invalid states.

**How to reproduce it (as minimally and precisely as possible)**:

I am not 100% sure how to reproduce this but I have unified alerts configured and I can see in the database that the state generated does not match any of the known states in the code.  I've tested 9.0.0, 9.0.1, and 9.0.2 and they all seem to have the same behavior.

I am not 100% sure which version may have introduced the issue with annotations being added with incorrect states but based on the timelines of when I was running specific versions, I would have been running `9.0.1` when the example annotation was added.

If it would helpful, I can try to figure out how to export the alert for review.  Just to note, these were migrated from legacy alerts to unified alerts in case that matters.

**Anything else we need to know?**:

**Environment**:
- Grafana version: `9.0.2`
- Data source type & version: `InfluxDB v2.2.0` - Alpine Linux (`influxdb:2.2-alpine` Docker image)
- OS Grafana is installed on: `grafana/grafana:9.0.2` Docker image
- User OS & Browser: macOS Monterey 12.4
- Grafana plugins: just the default plugins
- Others: n/a

**Comment by grobinson-grafana** [comment]
Hi! 👋 ,

This has been fixed and will be available in Grafana 9.0.3 (https://github.com/grafana/grafana/pull/51637)

**Comment by gillesdemey** [comment]
Closing this one as it's already fixed like @grobinson-grafana mentioned and available in the [9.0.3 release](https://github.com/grafana/grafana/releases/tag/v9.0.3). 

Thanks for the thorough bug report though – we rarely get  exemplary reports such as this one @mbentley ❤️
